Activities/Database/UiPath.Database.Activities/DatabaseExecute.cs

```diff
@@ -0,0 +1,121 @@
+using UiPath.Database.Activities.Properties;
+using System.Activities.Validation;
+
+namespace UiPath.Database.Activities
+{
+    public abstract class DatabaseExecute : AsyncTaskCodeActivity
+    {
+        private const int _defaultTimeout = 30000;
+
+        private Dictionary<string, Argument> parameters;
+
+        protected DatabaseConnection DbConnection = null;
+
+        [DefaultValue(null)]
+        [LocalizedCategory(nameof(Resources.ConnectionConfiguration))]
+        [LocalizedDisplayName(nameof(Resources.Activity_DatabaseExecute_Property_ProviderName_Name))]
+        [LocalizedDescription(nameof(Resources.Activity_DatabaseExecute_Property_ProviderName_Description))]
+        public InArgument<string> ProviderName { get; set; }
+
+        [LocalizedCategory(nameof(Resources.ConnectionConfiguration))]
+        [DefaultValue(null)]
+        [LocalizedDisplayName(nameof(Resources.Activity_DatabaseExecute_Property_ConnectionString_Name))]
+        [LocalizedDescription(nameof(Resources.Activity_DatabaseExecute_Property_ConnectionString_Description))]
+        public InArgument<string> ConnectionString { get; set; }
+
+        [DefaultValue(null)]
+        [LocalizedCategory(nameof(Resources.ConnectionConfiguration))]
+        [LocalizedDisplayName(nameof(Resources.Activity_DatabaseExecute_Property_ConnectionSecureString_Name))]
+        [LocalizedDescription(nameof(Resources.Activity_DatabaseExecute_Property_ConnectionSecureString_Description))]
+        public InArgument<SecureString> ConnectionSecureString { get; set; }
+
+        [LocalizedCategory(nameof(Resources.ConnectionConfiguration))]
+        [LocalizedDisplayName(nameof(Resources.Activity_DatabaseExecute_Property_ExistingDbConnection_Name))]
+        [LocalizedDescription(nameof(Resources.Activity_DatabaseExecute_Property_ExistingDbConnection_Description))]
+        public InArgument<DatabaseConnection> ExistingDbConnection { get; set; }
+
+        [DefaultValue(CommandType.Text)]
+        [LocalizedDisplayName(nameof(Resources.Activity_DatabaseExecute_Property_CommandType_Name))]
+        [LocalizedDescription(nameof(Resources.Activity_DatabaseExecute_Property_CommandType_Description))]
+        public CommandType CommandType { get; set; }
+
+        [DefaultValue(null)]
+        [LocalizedCategory(nameof(Resources.Input))]
+        [Browsable(true)]
+        [LocalizedDisplayName(nameof(Resources.Activity_DatabaseExecute_Property_Parameters_Name))]
+        [LocalizedDescription(nameof(Resources.Activity_DatabaseExecute_Property_Parameters_Description))]
+        public Dictionary<string, Argument> Parameters
+        {
+            get
+            {
+                if (this.parameters == null)
+                {
+                    this.parameters = new Dictionary<string, Argument>();
+                }
+                return this.parameters;
+            }
+            set
+            {
+                parameters = value;
+            }
+        }
```

Comment:
I know this is not actually your code, but please let's clean up this property:
- it's by default browsable anyway
- default value is actually new Dictionary
- the backing field can be removed (and its name is not following the convention)